.github/workflows/run_tests.yml

```diff
@@ -58,13 +58,15 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-20.04] # cannot test on macOS as docker isn't supported on Mac
+        os: [ubuntu-22.04] # cannot test on macOS as docker isn't supported on Mac
         rai_v: [1.2.7] # versions of RedisAI
-        py_v: ['3.7.x', '3.8.x', '3.9.x', '3.10.x'] # versions of Python
-        compiler: [intel, 8, 9, 10, 11] # intel compiler, and versions of GNU compiler
+        py_v: ['3.8.x', '3.9.x', '3.10.x'] # versions of Python
+        compiler: [nvhpc-23-11, intel, gcc-11, gcc-12] # intel compiler, and versions of GNU compiler
+        link_type: [Static, Shared]
```

Comment:
This doubles the number of runners needed, which in turn will make the CI/CD process take a much longer amount of time to complete since it will be pushing us beyond the available amount

Reply:
Good point, will follow up with the rest of the group